src/main/java/org/example/service/UserService.java

```diff
@@ -0,0 +1,106 @@
+package org.example.service;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.Persistence;
+import org.example.model.User;
+import org.example.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Service
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+
+    @Autowired
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public User saveUser(User user){
+        if(validatePassword(user.getPassword()) && validateEmail(user.getEmail())
+                && validateNames(user.getFirstname()) && validateNames(user.getLastname())){
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            return this.userRepository.save(user);
+        }
+
+        return null;
+    }
+
+    public User updateUser(User user){
+        if(validatePassword(user.getPassword())){
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            EntityManagerFactory entityManagerFactory = Persistence.createEntityManagerFactory("CatalogPersistence");
```

Comment:
do we need this for an update user?

Reply:
Afaik that's the only way to update a db entry, also requiring persistence.xml